### PR TITLE
Move infra to ts

### PR DESCRIPTION
**Pull request by sshmaxime**

### CODE REVIEW

File: hardhat.config.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                count: 10,
+                accountsBalance: '10000000000000000000000000000'
+            }
+        },
+
+        ...configNetworks
+    },
+
+    solidity: {
+        version: '0.6.12',
```
Comment (barakman) [inline hardhat.config.ts:45]: Perhaps we should read this during runtime from package.json, i.e., something like:
```
JSON.parse(fs.readFileSync('package.json')).devDependencies.solc
```

**Review by barakman — CHANGES_REQUESTED** [review]

File: hardhat.config.ts (hunk 1 above)
Comment (lbeder) [reply hardhat.config.ts:45]: I think it's overkill. We'd still have many other locations (solhint, pragmas, etc.) with an explicit version specified

Comment (barakman) [reply hardhat.config.ts:45]: Of course, first and foremost - the contracts themselves.
But at least we'll have this dependency slightly reduced.
I think it's worth the cost, unless it yields a performance hit upon starting the tests.

Comment (lbeder) [reply hardhat.config.ts:45]: It's not standard. BTW, keep in mind that the only reason we're installing `solc` in `package.json` is for solhint. Hardhat (and truffle, previously) download their own version of `solc`. So your suggestion might break things :(

**Review by lbeder — APPROVED** [review]

**Review by yudilevi — APPROVED** [review]

**Review by barakman — APPROVED** [review]

**Pull request merged**